intelmq/bots/parsers/stix/parser.py

```diff
@@ -0,0 +1,149 @@
+"""
+SPDX-FileCopyrightText: 2025 Ladislav Baco
+SPDX-License-Identifier: AGPL-3.0-or-later
+
+Parse indicators objects in STIX format received from TAXII collector
+"""
+
+import json
+
+
+from intelmq.lib.bot import ParserBot
+from intelmq.lib.exceptions import MissingDependencyError
+
+try:
+    import stix2patterns.v21.pattern as stix2_pattern
+except ImportError:
+    stix2_pattern = None
+
+
+class StixParserBot(ParserBot):
+    """Parse STIX indicators"""
+    parse = ParserBot.parse_json_stream
+    recover_line = ParserBot.recover_line_json_stream
+
+    def init(self):
+        if stix2_pattern is None:
+            raise MissingDependencyError('stix2-patterns')
+
+    def parse_line(self, line, report):
+        """ Parse one STIX object of indicator type """
+        object_type = line.get('type', '')
+        if object_type == 'indicator':
+            pattern = line.get('pattern', '')
+            # stix, pcre, sigma, snort, suricata, yara
+            pattern_type = line.get('pattern_type', '')
+
+            if pattern_type == 'stix':
+                indicators = StixParserBot.parse_stix_pattern(pattern, self.logger)
+                for indicator_type, indicator_value in indicators:
+                    event = self.new_event(report)
+                    event.add('raw', json.dumps(line))
```

Comment:
We can use the unparsed line by the json parser without the need of encoding it again:
```suggestion
                    event.add('raw', self.recover_line())
```